### ISSUE TITLE
Flag boolean expressions that must be true

### ISSUE THREAD

**Issue by filiph** [dart-lang/linter#178]
```dart
if (a == 0) {
  ...
  if (b.isEmpty && a == 0) {
    ...
  }
}
```

Although harmless in this case, the developer should still be warned about the irrelevant second check for `a == 0`. 

A more serious case would be `if (b.isEmpty || a == 0)`.

**Comment by alexeieleusis** [comment]
#229 addresses this issue, consensus is that the rule is too complex in its current state and will be splitted/simplified into a few different rules.

**Comment by alexeieleusis** [comment]
This might be linted now that https://github.com/dart-lang/linter/pull/229 is merged. I might add a test to confirm @pq can you assign this to me?

**Comment by pq** [comment]
Another one for you @alexeieleusis to self-assign once he's a collaborator proper. 😉
